File: sample/app/src/main/java/com/example/activitystarter/savetest/SavingActivity.java
package com.example.activitystarter.savetest;

import android.app.Activity;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.example.activitystarter.R;

import activitystarter.ActivityStarter;
import activitystarter.Arg;

public class SavingActivity extends Activity {

    public static final int DEFAULT_I = -1;
    public static final String DEFAULT_STR = "AAA";
    public static final boolean DEFAULT_B = false;

    public static final int NEW_I = 100;
    public static final String NEW_STR = "BBB";
    public static final boolean NEW_B = true;

    @Arg(optional = true) public int i = DEFAULT_I;
    @Arg(optional = true) public String str = DEFAULT_STR;
    @Arg(optional = true) public boolean b = DEFAULT_B;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityStarter.fill(this);
        setContentView(R.layout.activity_save_test);
        ((TextView) findViewById(R.id.i)).setText(""+i);
        ((TextView) findViewById(R.id.str)).setText(""+str);
        ((TextView) findViewById(R.id.b)).setText(""+b);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        ActivityStarter.save(this, outState);
    }
}

File: sample/kotlinapp/src/main/java/com/marcinmoskala/kotlinapp/savetest/SavingActivity.kt
package com.marcinmoskala.kotlinapp.savetest

import android.app.Activity
import android.os.Bundle
import android.widget.TextView

import com.marcinmoskala.kotlinapp.R

import activitystarter.ActivityStarter
import activitystarter.Arg
import com.marcinmoskala.activitystarter.argExtra
import kotlinx.android.synthetic.main.activity_save_test.*

class SavingActivity : Activity() {

    @get:Arg(optional = true) var i: Int by argExtra(NonSavingActivity.DEFAULT_I)
    @get:Arg(optional = true) var str: String by argExtra(NonSavingActivity.DEFAULT_STR)
    @get:Arg(optional = true) var b: Boolean by argExtra(NonSavingActivity.DEFAULT_B)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActivityStarter.fill(this, savedInstanceState)
        setContentView(R.layout.activity_save_test)
        iView.text = "$i"
        strView.text = str
        bView.text = "$b"
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        ActivityStarter.save(this, outState)
    }

    companion object {

        const val DEFAULT_I = -1
        const val DEFAULT_STR = "AAA"
        const val DEFAULT_B = false

        const val NEW_I = 100
        const val NEW_STR = "BBB"
        const val NEW_B = true
    }
}
